codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeParamsSpec.java

```diff
@@ -126,5 +140,40 @@ private void addBuilderSetterMethods(TypeSpec.Builder b) {
                                               "aws.auth#sigv4") // TODO: Reference the SigV4 AuthScheme when implemented
                                   .build());
         }
+
+        if (authSchemeSpecUtils.generateEndpointBasedParams()) {
+            parameters().forEach((name, model) -> {
+                if (authSchemeSpecUtils.includeParam(model, name)) {
+                    b.addMethod(setterMethodDeclaration(name, model));
+                }
+            });
+        }
+    }
+
+    private MethodSpec setterMethodDeclaration(String name, ParameterModel model) {
+        MethodSpec.Builder spec = MethodSpec.methodBuilder(authSchemeSpecUtils.paramMethodName(name));
+        spec.addModifiers(Modifier.PUBLIC);
+        if (model.getDeprecated() != null) {
+            spec.addAnnotation(Deprecated.class);
+        }
+        spec.addModifiers(Modifier.ABSTRACT)
+            .addParameter(parameterSpec(name, model))
+            .returns(authSchemeSpecUtils.parametersInterfaceBuilderInterfaceName());
+        if (model.getDocumentation() != null) {
+            spec.addJavadoc(model.getDocumentation());
+        }
+        return spec.build();
+    }
+
+    private ParameterSpec parameterSpec(String name, ParameterModel model) {
```

Comment:
nit: I'd probably still reuse the methods from EndpointRulesSpecUtils instead of duplicating here.